mantidproject/mantid
# Removing vms_compat dependency from LoadISISNexus

**Description of work.**

To load the new Muon Nexus files (V2) the LoadISISNexus2 loader will be utilised. In order to use this loader, the dependence of the LoadISISNexus2 on the isis_vms_compat was removed in this PR. Primarily this PR extracts a number of functions into a set of helper functions.

The following information was removed from the sample logs, as it was deemed either unnecessary, or was no longer being written to the nexus files.

This PR represents the first stage of developing the new Muon Nexus V2 loader #27701.

Variable | Code description|
|-------------------------|-----------|
run_header | Provides a brief description of the run. e.g instrument  and run number |
duration | Actual run duration
duration units | Scalar for run duration
duration_freq | Test interval for duration
dmp | Dump interval
dmp units | Scalar for dump interval
dmp_freq | Interval for dump
dur_wanted | Requested run duration
dur_secs | Actual run duration
mon_sum1 | Monitor sum1
mon_sum2 | Monitor sum2
mon_sum3 | Monitor sum3

**To test:** 
Outside of loading a workspace and checking the correct entries are there, you can
1. On a master branch load a workspace with LoadISISNexus and then save it
```
workspace_1 = LoadISISNexus("LOQ49886.nxs") 
SaveNexus(workspace_1,"LoadISISBefore.nxs")
```
2. Switch to this branch and load the same workspace with LoadISISNexus and compare the workspaces
```
workspace_1 = LoadISISNexus("LOQ49886.nxs") # load workspace with updated alg
workspace_2 = LoadNexus("LoadISISBefore.nxs") # load the workspace created from old alg
workspace_1.equals(workspace_2, 1e-9) # both should be equal
```
3. Additionally, the LoadISISNexus algorithm should now also load the Muon V2 file (you'll have to ask me for an example file)
4. Test that the Muon V2 file is successfully loaded, e.g plot some spectra, check sample logs.


Fixes #27695 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
